Superpowered/AndroidIO/SuperpoweredAndroidAudioIO.cpp

```diff
@@ -138,124 +433,140 @@ SuperpoweredAndroidAudioIO::SuperpoweredAndroidAudioIO(int samplerate, int buffe
-    };
+            (*internals->outputBufferQueue)->Realize(internals->outputBufferQueue, SL_BOOLEAN_FALSE);
+        };
 
-    if (enableOutput) { // Initialize the audio output buffer queue.
-        memset(internals->outputFifo.buffer, 0, internals->fifoBufferSizeBytes);
-        (*internals->outputBufferQueue)->GetInterface(internals->outputBufferQueue, SL_IID_BUFFERQUEUE, &internals->outputBufferQueueInterface);
-        (*internals->outputBufferQueueInterface)->RegisterCallback(internals->outputBufferQueueInterface, SuperpoweredAndroidAudioIO_OutputCallback, internals);
-        (*internals->outputBufferQueueInterface)->Enqueue(internals->outputBufferQueueInterface, internals->outputFifo.buffer, (SLuint32)buffersize * NUM_CHANNELS * 2);
-    };
+        if (enableInput) { // Initialize the audio input buffer queue.
+            memset(internals->inputFifo.buffer, 0, internals->fifoBufferSizeBytes);
+            (*internals->inputBufferQueue)->GetInterface(internals->inputBufferQueue, SL_IID_ANDROIDSIMPLEBUFFERQUEUE, &internals->inputBufferQueueInterface);
+            (*internals->inputBufferQueueInterface)->RegisterCallback(internals->inputBufferQueueInterface, SuperpoweredAndroidAudioIO_InputCallback, internals);
+            (*internals->inputBufferQueueInterface)->Enqueue(internals->inputBufferQueueInterface, internals->inputFifo.buffer, (SLuint32)buffersize * NUM_CHANNELS * 2);
+        };
+
+        if (enableOutput) { // Initialize the audio output buffer queue.
+            memset(internals->outputFifo.buffer, 0, internals->fifoBufferSizeBytes);
+            (*internals->outputBufferQueue)->GetInterface(internals->outputBufferQueue, SL_IID_BUFFERQUEUE, &internals->outputBufferQueueInterface);
+            (*internals->outputBufferQueueInterface)->RegisterCallback(internals->outputBufferQueueInterface, SuperpoweredAndroidAudioIO_OutputCallback, internals);
+            (*internals->outputBufferQueueInterface)->Enqueue(internals->outputBufferQueueInterface, internals->outputFifo.buffer, (SLuint32)buffersize * NUM_CHANNELS * 2);
+        };
 
-    startQueues(internals);
+        startQueues(internals);
+    }
 }
 
 void SuperpoweredAndroidAudioIO::onForeground() {
     internals->foreground = true;
-    startQueues(internals);
+    if (internals->aaudio) startAAudio(internals); else startQueues(internals);
 }
 
 void SuperpoweredAndroidAudioIO::onBackground() {
     internals->foreground = false;
 }
 
 void SuperpoweredAndroidAudioIO::start() {
-    startQueues(internals);
+    if (internals->aaudio) startAAudio(internals); else startQueues(internals);
 }
 
 void SuperpoweredAndroidAudioIO::stop() {
-    stopQueues(internals);
+    if (internals->aaudio) stopAAudio(internals); else stopQueues(internals);
 }
 
 void SuperpoweredAndroidAudioIO::resetInputBufferIndices() {
```

Comment:
This remains from the previous implementation to minimize latency (https://github.com/ResonantCavity/Vocoder/pull/220).

Reply:
👍